### ISSUE TITLE
[Android] Downloading an image into a directory that isn't created fails but works on iOS

### ISSUE THREAD

**Issue by dwilt** [wkh237/react-native-fetch-blob#215]
We are downloading some images in our application. It's working beautifully on iOS. Here is our `fetch`:

```javascript
export const storyAssetsPath = `${documentsDir}/storyAssets`;

const downloadImage = async url => {
    const id = uuid();
    const filename = `${id}.jpg`;
    const path = `${storyAssetsPath}/${filename}`;

    await RNFetchBlob.config({
        fileCache: true,
        path
    }).fetch('GET', url);

    return filename;
};
```
Notice the `storyAssets` folder that we're downloading the image into.

On Android, if the folder *does not* exist, it will not download the images correctly. It will seemingly succeed as the `res` comes back from the `.fetch` successfully but the images aren't actually there. If I manually create the folder on the device:

```shell
adb shell
root@vbox86p:/ $run-as com.greatjonesstreet
root@vbox86p:/ $ cd /data/data/com.greatjonesstreet
root@vbox86p:/data/data/com.greatjonesstreet $ mkdir storyAssets
```
and run it again, the images download correctly. Any ideas?

`react-native`: `0.37.0`
`react-native-fetch-blob`: `0.10.0`

**Comment by wkh237** [comment]
@dwilt , thanks for reporting this issue. Looks like on Android it does not create folder automatically, as an alternative, you can use [fs.exists](https://github.com/wkh237/react-native-fetch-blob/wiki/File-System-Access-API#existspathstringpromise) and [fs.mkdir](https://github.com/wkh237/react-native-fetch-blob/wiki/File-System-Access-API#mkdirpathstringpromise) to check and create the folder before download started.

**Comment by dwilt** [comment]
Yep, that's what I did. Should probably do that anyway - it's more explicit.

**Comment by colorfulberry** [comment]
@dwilt , I have suffered same things. Have you find a solution about this ?

**Comment by dwilt** [comment]
@colorfulberry I found a workaround, like @wkh237 suggested too. Here's what I'm doing:

```javascript
const assetsDirExists = await RNFetchBlob.fs.isDir(storyAssetsPath);

if (!assetsDirExists) {
    await RNFetchBlob.fs.mkdir(storyAssetsPath);
}
```
I first check to se if that path exists. if it doesn't, I create it.

**Comment by colorfulberry** [comment]
@dwilt  Thanks. 🎄
